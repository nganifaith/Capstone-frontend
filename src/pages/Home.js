import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { setBeaches } from '../actions';
import { beachesData } from '../Api';
import Main from '../components/Main';
import Beach from '../components/Beach';
import SearchBar from '../components/SearchBar';
import { getBeaches } from '../reducers/beachesList';

const Home = () => {
  const [searchTerm, setSearchTerm] = useState('');
  const [error, setError] = useState('');
  const dispatch = useDispatch();

  const beaches = useSelector(getBeaches);
  useEffect(() => {
    setError('');
    beachesData().then((beaches) => {
      dispatch(setBeaches(beaches));
    });
  }, [searchTerm]);

  return (
    <div>
      <SearchBar setSearchTerm={setSearchTerm} />
      <span>{error}</span>
      <Main header={searchTerm ? 'Search Results' : 'All Sites'}>
        {beaches.map((beach) => (
          <Beach key={beach.id} beach={beach} />
        ))}
      </Main>
    </div>
  );
};

export default Home;
