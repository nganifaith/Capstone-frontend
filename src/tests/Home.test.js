import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createStore } from 'redux';
import rootReducer from '../reducers';

import Home from '../pages/Home';

const history = createMemoryHistory();
const store = createStore(rootReducer);

test('full app rendering/navigating', () => {
  const home = renderer.create(
    <Provider store={store}>
      <Router history={history}>
        <Home />
      </Router>
      ,
    </Provider>,
  ).toJSON();
  expect(home).toMatchSnapshot();
});

it('renders banner', () => {
  render(
    <Provider store={store}>
      <Router history={history}>
        <Home />
      </Router>
    </Provider>,
  );
  const listings = screen.getByRole('heading');
  expect(listings).toBeInTheDocument();
});
