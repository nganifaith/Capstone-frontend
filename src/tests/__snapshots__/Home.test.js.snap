// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full app rendering/navigating 1`] = `
Array [
  <div>
    <span>
      
    </span>
    <div
      className="sc-bdfBQB lfOvXU"
    >
      <h1>
        Listings -
         
        All listings
      </h1>
      <div
        className="sc-gsTEea ipeXrw"
      />
    </div>
  </div>,
  ",",
]
`;
